### PR TITLE
helix: fix UB in diff gutter

### PR DESCRIPTION
**Pull request by lunagl**
## Description of changes

Applies https://github.com/helix-editor/helix/pull/7227 as a patch until the fix is included in the next release.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by atalii** [comment]
should get a review from @danth @yusdacra or @zowoq. my words carry little weight, but i'm on board with the "just get UB out the tree now and fix it later" approach so long as the package maintainers are.

other than that, lgtm after a local build.

**Review by atalii — APPROVED** [review]
i'm approving for now barring any other opinions

**Comment by fschwalbe** [comment]
Yeah I was hoping there would be a new helix release that fixes this sooner, but as I'm not sure how long it will take until then (https://github.com/helix-editor/helix/issues/7896#issuecomment-1677393131 says it could be sometime this or next month), I thought It'd be best to temporarily patch it here ourselves.

It's not just that there's UB, the bug actively prevents me from using helix on any committed files in a git repo which is kinda annoying.

**Comment by yusdacra** [comment]
Result of `nixpkgs-review pr 249163` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>helix</li>
  </ul>
</details>

**Review by 90-008 — COMMENTED** [review]
Tested locally, seems to work as intended. LGTM

**Review by 90-008 — APPROVED** [review]

File: pkgs/applications/editors/helix/default.nix
Comment (zowoq) [inline pkgs/applications/editors/helix/default.nix:18]: Is there a reason you're not using `fetchpatch`?

Comment (fschwalbe) [reply pkgs/applications/editors/helix/default.nix:18]: I didn't know `fetchpatch` existed, should I change it to use that?

**Comment by beeb** [comment]
Fixes the issue for me too, LGTM

**Pull request merged**
